tests/pixelpipeline/test_datasourcefactories.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+import numpy as np
+
+from numpy.random import rand
+
+from volumina.pixelpipeline import datasourcefactories as factories
+from volumina.pixelpipeline import datasources as ds
+
+
+@pytest.fixture
+def lazyflow_op():
+    graph = pytest.importorskip("lazyflow.graph")
+
+    class OpPiper(graph.Operator):
+
+        Input = graph.InputSlot()
+        Output = graph.OutputSlot()
+
+        def setupOutputs(self):
+            self.outputs["Output"].meta.assignFrom(self.inputs["Input"].meta)
+            self.outputs["Output"].connect(self.inputs["Input"])
+
+        def execute(self, slot, subindex, roi, result):
+            result[:] = self.outputs["Output"](roi).wait()
+            return result
+
+        def propagateDirty(self, inputSlot, subindex, roi):
+            self.Output.setDirty(roi)
+
+    g = graph.Graph()
+    op = OpPiper(graph=g)
+
+    return op
+
+
+@pytest.fixture
+def vigra():
+    return pytest.importorskip("vigra")
+
+
+@pytest.mark.parametrize("ndims", range(2, 6))
+def test_lazyflow_factory(lazyflow_op, vigra, ndims):
+    shape = (10,) * ndims
+
+    array = rand(*shape).view(vigra.VigraArray)
+    array.axistags = vigra.defaultAxistags("txyzc"[:ndims])
+    lazyflow_op.inputs["Input"].setValue(array)
+
+    src = factories.createDataSource(lazyflow_op.Input)
+    assert isinstance(src, ds.LazyflowSource)
+    assert np.squeeze(np.ndarray(src._op5.Output.meta.shape)).shape == array.shape
```

Comment:
hmmm you're just checking the shapes, which will only contain `10`s, so actually you could just compare the length. It would however be a stronger comparison, if shapes in the respective axes differ, so one can see that order is preserved, too.
Also while one is at it, one could check that the data is not altered, too (`numpy.testing.assert_array_equal...`)